vespalib/src/vespa/vespalib/util/threadstackexecutorbase.h

```diff
@@ -78,16 +80,20 @@ class ThreadStackExecutorBase : public SyncableThreadExecutor,
 
     struct BlockedThread {
         const uint32_t wait_task_count;
-        Monitor monitor;
+        mutable std::mutex monitor;
+        mutable std::condition_variable cond;
         bool blocked;
         BlockedThread(uint32_t wait_task_count_in)
-            : wait_task_count(wait_task_count_in), monitor(), blocked(true) {}
+            : wait_task_count(wait_task_count_in), monitor(), cond(), blocked(true) {}
         void wait() const;
         void unblock();
     };
 
     std::unique_ptr<FastOS_ThreadPool>   _pool;
-    Monitor                              _monitor;
+    mutable std::mutex                   _monitor;
+protected:
+    std::condition_variable              _cond;
```

Comment:
prefer passing this in addition to the held lock in wakeup to avoid protected fields

Reply:
Done